src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -63,29 +72,54 @@ public JwtTokenReader(TokenValidationParameters tokenValidationParameters, strin
         /// <param name="token">The JWT token.</param>
         public async Task<bool> IsValidTokenAsync(string token)
         {
+            bool result;
+
             try
             {
-                TokenValidationParameters validationParameters = await DetermineTokenValidationParametersAsync();
-                _handler.ValidateToken(token, validationParameters, out SecurityToken jwtToken);
+                var validationParameters = await DetermineTokenValidationParametersAsync();
+                _handler.ValidateToken(token, validationParameters, out var validatedToken);
+                result = ValidateClaimCheck(validatedToken);
             }
-            catch (Exception exception)
+            catch
             {
                 return false;
             }
 
-            return true;
+            return result;
+        }
+
+        /// <summary>
+        /// Verifies if the claims coming in ClaimCheck object vs the claims in the jwt object based on the claim name and value are valid
+        /// </summary>
+        /// <param name="validatedToken">Security token which contains the jwt claims</param>
+        /// <returns></returns>
+        public bool ValidateClaimCheck(SecurityToken validatedToken)
+        {
+            if (_claimCheck == null || !_claimCheck.Any())
+            {
+                return true;
+            }
+
+            var jwtToken = (JwtSecurityToken) validatedToken;
```

Comment:
A type cast exception can happen here. If we only allow such a type, we should `Guard` here.

Reply:
done